src/main/java/com/google/sps/servlets/GetBlurAreasServlet.java

```diff
@@ -0,0 +1,211 @@
+public class GetBlurAreasServlet extends HttpServlet {
+
+  // Image types that are supported by our application.
+  private static final ArrayList<String> supportedTypes =
+      new ArrayList<>(Arrays.asList("image/jpeg", "image/png"));
+
+  /**
+   * This method handles the POST requests to "/get-blur-areas". Receives a BlobKey parameter which
+   * represents the Blobstore location of an image. Responds with a JSON ArrayList of rectangles. A
+   * rectangle is represented by a List of points. A point contains two properties, its coordinates:
+   * x and y.
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the BlobKey that points to the image uploaded by the user.
+    BlobKey blobKey = getBlobKey(request, "image");
+
+    // User didn't upload a file, so render an error message.
+    if (blobKey == null) {
+      response.setContentType("text/html;");
+      response.getWriter().println("Please upload an image file.");
+      response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+      return;
+    }
+
+    // User uploaded an unsupported type of file, so render an error message.
+    if (!isImageSupported(blobKey)) {
+      deleteFile(blobKey);
+
+      response.setContentType("text/html;");
+      response.getWriter().println("Image extension not supported.");
+      response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+      return;
+    }
+
+    // Get the image the user uploaded as bytes.
+    byte[] imageBytes = getBlobBytes(blobKey);
+
+    ArrayList<List<Point>> blurAreas = getBlurAreas(imageBytes);
+    deleteFile(blobKey);
+
+    // Convert the rectangles to JSON.
+    Gson gson = new Gson();
+    String jsonResponse = gson.toJson(blurAreas);
+
+    // Send the JSON back as the response.
+    response.setContentType("application/json");
+    response.getWriter().println(jsonResponse);
+  }
+
+  /**
+   * Uses the Google Cloud Vision API to find faces in the image represented by the binary data
+   * stored in @param imageBytes
+   *
+   * @return an ArrayList of bounding rectangles representing faces. rectangles are represented by a
+   *     list of points.
+   */
+  private ArrayList<List<Point>> getBlurAreas(byte[] imageBytes) throws IOException {
+    // This is the array that we will return.
+    ArrayList<List<Point>> facePolys = new ArrayList<>();
+
+    // Convert bytes to an Image object.
+    ByteString byteString = ByteString.copyFrom(imageBytes);
+    Image image = Image.newBuilder().setContent(byteString).build();
+
+    // Create a face detection request to our image.
+    Feature feature = Feature.newBuilder().setType(Feature.Type.FACE_DETECTION).build();
+    AnnotateImageRequest request =
+        AnnotateImageRequest.newBuilder().addFeatures(feature).setImage(image).build();
+
+    // Add request to an array.
+    List<AnnotateImageRequest> requests = new ArrayList<>();
+    requests.add(request);
```

Comment:
This move is not clear. I guess you need it because client.batchAnnotateImages takes an array, but you can explain it in the comment to make it more clear.

Reply:
Done, thanks! :)